cilib/models/repos/__init__.py

```diff
@@ -15,6 +19,40 @@ def __init__(self, repo=None, git_user=None, name=None):
     def __str__(self):
         return self.repo
 
+    def cat(self, branch, path):
+        """Cat file from a git repo"""
+        parsed = urlparse(self.repo)
+        if "git.launchpad.net" in parsed.netloc:
+            """
+            Launchpad supports viewing files directly from the web interface
+            for example)
+                repo: git+ssh://k8s-team-ci@git.launchpad.net/snap-kube-apiserver
+                branch: v1.28.13
+                file: /snapcraft.yaml
+
+                becomes
+                https://k8s-team-ci@git.launchpad.net/snap-kubectl/plain/snapcraft.yaml?h=v1.28.13
+            """
+            full_path = Path("/") / path
+            url = f"https://{parsed.netloc}{parsed.path}/plain{full_path}?h={branch}"
+        elif "github.com" in parsed.netloc:
+            """
+            Github supports viewing files directly from the web interface
+            https://raw.githubusercontent.com/kubernetes/kubernetes/refs/tags/v1.31.0/.go-version
+            """
+            full_path = Path("/") / path
+            url = f"https://raw.githubusercontent.com{parsed.path}/refs/tags/{branch}{full_path}"
+        else:
+            raise NotImplementedError("Only launchpad.net and github.com supported")
+
+        response = requests.get(url)
+        if response.status_code == 200:
+            return response.text
+        elif response.status_code == 404:
+            raise FileNotFoundError(f"File not found: {url}")
+        else:
+            raise Exception(f"Failed to fetch {url}: {response.status_code}")
+
```

Comment:
Implements a way to get the contents of a file hosted on git without cloning the whole repo.  Fortunately public launchpad and github projects support raw file content download like this